scripts/zkai_larger_native_boundary_candidate_selector_gate.py

```
@@ -0,0 +1,779 @@
+        candidate.resolve().relative_to(ROOT.resolve())
+    except ValueError as err:
+        raise LargerNativeBoundaryCandidateSelectorError(f"{label} escapes repo root") from err
+    return candidate
+
+
+def verified_mlp_accounting(accounting_rows: dict[str, dict[str, Any]]) -> dict[str, int]:
+    row = accounting_rows.get(MLP_FUSED_ENVELOPE)
+    if row is None:
+        raise LargerNativeBoundaryCandidateSelectorError(
+            f"missing accounting row for {MLP_FUSED_ENVELOPE}"
+        )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)